source/frontend/src/features/mapSideBar/acquisition/tabs/agreement/form/AcquisitionAgreementForm.tsx

```diff
@@ -0,0 +1,148 @@
+import { FormikProps, getIn } from 'formik';
+import { useEffect } from 'react';
+import styled from 'styled-components';
+
+import { FastCurrencyInput, FastDatePicker, Input, Select } from '@/components/common/form';
+import { Section } from '@/components/common/Section/Section';
+import { SectionField, StyledFieldLabel } from '@/components/common/Section/SectionField';
+import * as API from '@/constants/API';
+import useLookupCodeHelpers from '@/hooks/useLookupCodeHelpers';
+import { useModalContext } from '@/hooks/useModalContext';
+import { ApiGen_CodeTypes_AgreementStatusTypes } from '@/models/api/generated/ApiGen_CodeTypes_AgreementStatusTypes';
+import { ApiGen_CodeTypes_AgreementTypes } from '@/models/api/generated/ApiGen_CodeTypes_AgreementTypes';
+import { mapLookupCode } from '@/utils/mapLookupCode';
+
+import { AcquisitionAgreementFormModel } from '../models/AcquisitionAgreementFormModel';
+import { StyledSectionSubheader } from '../styles';
+
+export interface IAcquisitionAgreementFormProps {
+  formikProps: FormikProps<AcquisitionAgreementFormModel>;
+}
+
+const AcquisitionAgreementForm: React.FunctionComponent<
+  React.PropsWithChildren<IAcquisitionAgreementFormProps>
+> = ({ formikProps }) => {
+  const { setModalContent, setDisplayModal } = useModalContext();
+  const { getOptionsByType, getByType } = useLookupCodeHelpers();
+
+  const agreementStatusOptions = getOptionsByType(API.AGREEMENT_STATUS_TYPES);
+  const agreementTypeOptions = getByType(API.AGREEMENT_TYPES);
+
+  const agreementStatusTypeCodeValue = getIn(formikProps.values, 'agreementStatusTypeCode');
+  const agreementTypeCodeValue = getIn(formikProps.values, 'agreementTypeCode');
+  const agreementCancellationNoteValue = getIn(formikProps.values, 'cancellationNote');
+  const agreementStatusTypeCodeTouched = getIn(formikProps.touched, 'agreementStatusTypeCode');
+
+  useEffect(() => {
+    if (
+      agreementStatusTypeCodeValue !== ApiGen_CodeTypes_AgreementStatusTypes.CANCELLED &&
+      !!agreementCancellationNoteValue
```

Comment:
unsure if this is safe. If Id isValidId, if string isValidString, if object exists()

Reply:
not sure what you mean here. I haven't changed the logic from the previous implementation.

Reply:
its about the double bang (!!) used here. Javascript has a bunch of quirks and it would be better if we move away from using them when possible. If this is just moved logic no need to update it.